soundbay/data.py

```diff
@@ -80,18 +81,16 @@ def _preprocess_metadata(self, len_buffer, equalize=False, slice_flag=False):
         """
 
         def _equalize_distribution(df_object):
-            len_pos = len(df_object[df_object['label'] == 1])
-            len_neg = len(df_object[df_object['label'] == 0])
-            diff = len_pos - len_neg
-            if diff == 0:
-                return df_object
-            label = 0 if diff > 0 else 1
-            multiplier = int(np.ceil(abs(diff) / min(len_neg, len_pos)))
-            additive = pd.concat(deepcopy([self.metadata[self.metadata['label'] == label]]) * multiplier)
-            additive = additive[:abs(diff)]
-            df_object = pd.concat([df_object, additive])
-            assert len(df_object[df_object['label'] == 1]) == len(df_object[df_object['label'] == 0])
-            return df_object
+            # TODO: How should we consider background?
+            oversample = RandomOverSampler(sampling_strategy='auto')
```

Comment:
Nice find. In the future perhaps we'll want an option to pass the input to this parameter from conf (to enable un-balanced sampling with a dict for e.g.)